deploy-service/universal/src/main/java/com/pinterest/teletraan/universal/security/BasePastisAuthorizer.java

```diff
@@ -0,0 +1,108 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pinterest.teletraan.universal.security;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.pinterest.commons.pastis.PastisAuthorizer;
+import com.pinterest.teletraan.universal.security.bean.AuthZResource;
+import com.pinterest.teletraan.universal.security.bean.TeletraanPrincipal;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import javax.ws.rs.container.ContainerRequestContext;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * A base class for authorizers that use Pastis for authorization. Pinterest only
+ *
+ * @param <P> the principal type
+ */
+@Slf4j
+public class BasePastisAuthorizer<P extends TeletraanPrincipal> extends BaseAuthorizer<P> {
+    private static final String INPUT = "input";
+    protected final PastisAuthorizer pastis;
+
+    @Builder
+    private BasePastisAuthorizer(
+            String serviceName, PastisAuthorizer pastis, AuthZResourceExtractor.Factory factory) {
+        super(factory);
+        if (pastis == null) {
+            if (serviceName == null) {
+                throw new IllegalArgumentException(
+                        "PastisAuthorizer and serviceName cannot both be null");
+            }
+            this.pastis = new PastisAuthorizer(serviceName);
+        } else {
+            this.pastis = pastis;
+        }
+    }
+
+    public BasePastisAuthorizer(PastisAuthorizer pastis, AuthZResourceExtractor.Factory factory) {
+        super(factory);
+        this.pastis = pastis;
+    }
+
+    @Override
+    public boolean authorize(
+            P principal,
+            String role,
+            AuthZResource requestedResource,
+            @Nullable ContainerRequestContext context) {
+
+        Map<String, PastisRequest> payload = new HashMap<>();
+        payload.put(INPUT, new PastisRequest(principal, role, requestedResource));
+        try {
+            String pastisPayload = new ObjectMapper().writeValueAsString(payload);
+            boolean authorized = pastis.authorize(pastisPayload);
+            log.debug("Authorized: {}", authorized);
+            return authorized;
+        } catch (Exception ex) {
+            log.debug("Failed to authorized via Pastis", ex);
+            // TODO: add fallback
```

Comment:
Create Jira task for this and remove TODO.

Reply:
Made it a method for subclasses to override.